src/config.rs

```diff
@@ -68,6 +73,72 @@ impl Config {
     }
 }
 
+#[derive(Serialize, Deserialize)]
+pub struct SubProcess {
+    pub command: String,
+    pub working_dir: Option<String>,
+    #[serde(default)]
+    pub keep_on_exit: bool,
+    #[serde(default)]
+    pub stdout: bool,
+    #[serde(default)]
+    pub stderr: bool,
+    #[serde(default)]
+    pub foreground: bool,
+}
+
+impl SubProcess {
+    pub fn spawn(&self, tmp_dir: &Path) -> Option<process::Child> {
+        let current_dir = if let Some(working_dir) = &self.working_dir {
+            Path::new(working_dir)
+        } else {
+            tmp_dir
+        };
+
+        let mut process = process::Command::new(run::get_shell());
+
+        process.current_dir(&current_dir);
+
+        #[cfg(unix)]
+        process.arg("-c");
+        #[cfg(windows)]
+        process.arg("/k");
+
+        if !self.stdout {
+            process.stdout(process::Stdio::null());
+        }
+
+        if !self.stderr {
+            process.stderr(process::Stdio::null());
+        }
+
+        process.arg(self.command.clone());
+
+        if self.foreground {}
+
+        if !self.foreground {
+            match process.arg(self.command.clone()).spawn().ok() {
```

Comment:
And here you add the command again in argument. I assume you didn't review yourself. I thought you said it was ready for me to review :disappointed: